plugins/custom-auth/index.js

```diff
@@ -0,0 +1,113 @@
+const Users = require('services/users');
+const debug = require('debug')('talk-plugin-custom-auth');
+const authz = require('middleware/authorization');
+const { get } = require('lodash');
+
+module.exports = {
+
+  // When the JWT is passed to Talk, first it will be validated
+  // Next, Talk will attempt to locate a user in its DB using the token's sub claim,
+  // if the user is not found, the tokenUserNotFound hook will be called to create the user
+  tokenUserNotFound: async ({ jwt }) => {
+    debug('The jwt is ' + JSON.stringify(jwt));
+    
+    // Since the JWT has already been validated we can pass it's claims directly to upsertExternalUser
+    const user = await Users.upsertExternalUser(
+      null,
```

Comment:
That was your suggestion originally, but that does not work. Requiring context as you suggested will throw `Cannot use 'in' operator to search for 'tags' in undefined`, but passing NULL works because `upserUser` only requires context to emit the user creation. It works fine without it.